core/storage/app_feedback_report/gae_models.py

```diff
@@ -0,0 +1,591 @@
+    app settings, as well as information provided by the user in the feedback
+    report.
+
+    The id of each model instance is determined by concatenating the platform,
+    the timestamp of the reports (in ms since epoch, in UTC), and a hash of a
+    string representation of a random int.
+    """
+
+    # We use the model id as a key in the Takeout dict.
+    ID_IS_USED_AS_TAKEOUT_KEY = True
+
+    # The platform (web or Android) that the report is sent from and that the
+    # feedback corresponds to.
+    platform = datastore_services.StringProperty(
+        required=True, indexed=True, choices=PLATFORM_CHOICES)
+    # The ID of the user that scrubbed this report, if it has been scrubbed.
+    scrubbed_by = datastore_services.StringProperty(
+        required=False, indexed=True)
+    # Unique ID for the ticket this report is assigned to (see
+    # AppFeedbackReportTicketModel for how this is constructed). This defaults
+    # to None since initially, new reports received will not be assigned to a
+    # ticket.
+    ticket_id = datastore_services.StringProperty(default=None, indexed=True)
+    # Datetime in UTC of when the report was submitted by the user on their
+    # device. This may be much earlier than the model entity's creation date if
+    # the report was locally cached for a long time on an Android device.
+    submitted_on = datastore_services.DateTimeProperty(
+        required=True, indexed=True)
+    # The type of feedback for this report; this can be an arbitrary string
+    # since iterations of the report structure may introduce new types and we
+    # cannot rely on the backend updates to fully sync with the frontend report
+    # updates.
+    report_type = datastore_services.StringProperty(required=True, indexed=True)
+    # The category that this feedback is for.
+    category = datastore_services.StringProperty(required=True, indexed=True)
+    # The version of the app; on Android this is the package version name (e.g.
+    # 1.0-release-arm...) and on web this is the release version (e.g. 3.0.8).
+    platform_version = datastore_services.StringProperty(
+        required=True, indexed=True)
+    # The user's country locale represented as a ISO-3166 code; the locale is
+    # determined by the user's Android device settings.
+    device_country_locale_code = datastore_services.StringProperty(
+        required=True, indexed=True)
+    # The entry point location that the user is accessing the feedback report
+    # from on both web & Android devices. On Android, this could be
+    # navigation_drawer, lesson_player, revision_card, or crash.
+    entry_point = datastore_services.StringProperty(required=True, indexed=True)
+    # Additional topic / story / exploration IDs that may be collected depending
+    # on  the entry_point used to send the report; lesson player will have
+    # topic_id, story_id, and exploration_id, while revision card while have
```

Comment:
``while revision cards will have`` (a few typos here & elsewhere in this property's doc string--I suggest fixing them).

Reply:
Done